bin/tx-prover/README.md

```diff
@@ -1,84 +1,89 @@
 # Miden transaction prover
 
-A service for generating Miden transaction proofs on-demand. It is split in two binaries: worker and proxy.
+A service for generating Miden transaction proofs on-demand. It includes a CLI to control both workers and proxy for the Miden's remote transaction prover service, and a client, `RemoteTransactionProver` to interact with the service from a Rust codebase.
 
 The worker is a gRPC service that can receive transaction witnesses and returns the proof. It can only handle one request at a time and returns an error if is already in use.
 
 The proxy uses [Cloudflare's Pingora crate](https://crates.io/crates/pingora), which provides features to create a modular proxy. It is meant to handle multiple workers with a queue for each one. Further information about Pingora and it's features can be found in the [official GitHub repository](https://github.com/cloudflare/pingora).
 
 ## Installation
 
-Install the prover worker and proxy binaries for production using `cargo`:
+<!-- The following documentation is documented since the CLI is not on crates-io yet -->
+<!-- Install the CLI for production using `cargo`:
 
 ```sh
-cargo install miden-tx-prover-worker --locked
-cargo install miden-tx-prover-proxy --locked
+cargo install miden-tx-prover --locked
 ```
 
 This will install the latest official version of the prover. You can install a specific version using `--version <x.y.z>`:
 
 ```sh
-cargo install miden-tx-prover-worker --locked --version x.y.z
-cargo install miden-tx-prover-proxy --locked --version x.y.z
-```
-
-Both services can be installed at once from the source code using specific git revisions with `cargo install`. Note that since these aren't official releases we cannot provide much support for any issues you run into, so consider this for advanced users only.
+cargo install miden-tx-prover --locked --version x.y.z
+``` -->
 
-If you want to build from a local version, from the root of the workspace you can run:
+To build the CLI from a local version, from the root of the workspace you can run:
 
 ```bash
-make install-prover-worker
-make install-prover-proxy
+make install-tx-prover
 ```
 
+The CLI can be installed from the source code using specific git revisions with `cargo install`. Note that since these aren't official releases we cannot provide much support for any issues you run into, so consider this for advanced users only.
+
 Note that for the prover worker you might need to enable the `testing` feature in case the transactions were executed with reduced proof-of-work requirements (or otherwise, the proving process will fail). This step will also generate the necessary protobuf-related files. You can achieve that by generating the binary with:
 
 ```bash
-make install-prover-worker-testing
+make install-tx-prover-testing
 ```
 
-## Running the Service
+## Usage
```

Comment:
nit (feel free to disregard): I think for readability it might be better to introduce workers here first, and then mention you can run a proxy against a number of running workers (maybe could also be separated in sections). But I wouldn't change the wording too much, mostly ordering.

Reply:
That actually sounds better.